File: 2126-destroying-asteroids/2126-destroying-asteroids.go
func asteroidsDestroyed(mass int, asteroids []int) bool {
    sort.Ints(asteroids)
    for i := 0; i < len(asteroids); i++ {
        if mass >= asteroids[i] {
            mass += asteroids[i]
        } else {
            return false
        }
    }

    return true
}

// array, sorting
// time: O(nlog(n))
// space: O(1)

File: 2126-destroying-asteroids/2126-destroying-asteroids.py
class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:
        asteroids.sort()
        for a in asteroids:
            if mass >= a:
                mass += a
            else:
                return False
        
        return True

# array, sorting
# time: O(nlog(n))
# space: O(1)